File: src/viewModels/EmployeeViewModels.ts
import { Employee } from '../domain/models/Employee';

interface IEmployeeViewModel extends Omit<Employee, 'creationDate' | 'modificationDate' | 'id'> { }

export class EmployeeViewModel implements IEmployeeViewModel {  
  id: number | null = 0;
  active: number = 0;
  firstName: string = '';
  lastName: string = '';
  hireDate: Date | null = null;
  phone: string | null = null;
  address: string | null = null;
  avatarUrl: string | null = null;
  departmentId: number | null = null;
  departmentName?: string | null = null;
}

File: src/wsample.api/ViewModels/EmployeeViewModels.cs
namespace wsample.api.ViewModels
{
    public class EmployeeViewModel
    {
        public int Id { get; set; }
        public bool Active { get; set; }
        public string FirstName { get; set; } = string.Empty;
        public string LastName { get; set; } = string.Empty;
        public DateTime? HireDate { get; set; }
        public string? Phone { get; set; }
        public string? Address { get; set; }
        public string? AvatarUrl { get; set; }
        public int? DepartmentId { get; set; }
        public string? DepartmentName { get; set; }
    }
}
